server/src-lib/Hasura/GraphQL/Transport/WebSocket.hs

```diff
@@ -210,23 +211,36 @@ sendMsg :: (MonadIO m) => WSConn -> ServerMsg -> m ()
 sendMsg wsConn msg =
   liftIO $ WS.sendMsg wsConn $ WS.WSQueueResponse (encodeServerMsg msg) Nothing
 
+-- sendCloseWithMsg closes the websocket server with an error code that can be supplied as (Maybe Word16),
+-- if there is `Nothing`, the server will be closed with an error code derived from ServerErrorCode
 sendCloseWithMsg ::
   (MonadIO m) =>
   L.Logger L.Hasura ->
   WSConn ->
   ServerErrorCode ->
   Maybe ServerMsg ->
+  Maybe Word16 ->
   m ()
-sendCloseWithMsg logger wsConn errCode mErrServerMsg = do
+sendCloseWithMsg logger wsConn errCode mErrServerMsg mCode = do
   case mErrServerMsg of
     Just errServerMsg -> do
       sendMsg wsConn errServerMsg
     Nothing -> pure ()
   logWSEvent logger wsConn EClosed
-  liftIO $ WS.sendClose wsc errMsg
+  liftIO $ WS.sendCloseCode wsc errCloseCode errMsg
   where
     wsc = WS.getRawWebSocketConnection wsConn
     errMsg = encodeServerErrorMsg errCode
+    errCloseCode = fromMaybe (getErrCode errCode) mCode
+    getErrCode :: ServerErrorCode -> Word16
+    getErrCode err = case err of
+      ProtocolError1002 -> 1002
+      GenericError4400 _ -> 4400
+      Unauthorized4401 -> 4401
+      Forbidden4403 -> 4403
+      ConnectionInitTimeout4408 -> 4408
+      NonUniqueSubscription4409 _ -> 4409
+      TooManyRequests4429 -> 4429
```

Comment:
Close codes are mentioned in [the reference sheet here](https://github.com/enisdenjo/graphql-ws/blob/master/docs/enums/common.CloseCode.md).

Reply:
Same here. This too, is a note right?

Reply:
Yupp